upload-current.sh

```diff
@@ -1,77 +1,89 @@
 
 MODEL_DIR=data/minio/bucket/current/model/
 
-while getopts ":c:" opt; do
+echo "Uploading to model ==>  s3://$S3_BUCKET/$S3_PREFIX  <=="
+
+USE_BEST=false
+while getopts ":c:b" opt; do
 case $opt in
-c) CHECKPOINT="$OPTARG"
-;;
-\?) echo "Invalid option -$OPTARG" >&2
-;;
+    c) CHECKPOINT="$OPTARG"
+       ;;
+    b) USE_BEST=true
+       ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+        ;;
 esac
 done
 
 CHECKPOINT_FILE=$MODEL_DIR"deepracer_checkpoints.json"
 if [ ! -f ${CHECKPOINT_FILE} ]; then
   echo "Checkpoint file not found!"
   exit 1
-else
-  echo "found checkpoint index file "$CHECKPOINT_FILE
-fi;
+fi
+echo "found checkpoint index file "$CHECKPOINT_FILE
 
 if [ -z "$CHECKPOINT" ]; then
-  echo "Checkpoint not supplied, checking for latest checkpoint"
-
-  LAST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".last_checkpoint.name"`
-  BEST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".best_checkpoint.name"`
-  
-  CHECKPOINT=$LAST_CHECKPOINT
-
-  echo "latest checkpoint = "$CHECKPOINT
+  #echo "Checkpoint not supplied, checking for latest checkpoint"
+  LAST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".last_checkpoint.name" | sed s/\"//g`
+  BEST_CHECKPOINT=`cat $CHECKPOINT_FILE |jq ".best_checkpoint.name" | sed s/\"//g`
+  if $USE_BEST; then
+     CHECKPOINT=$BEST_CHECKPOINT
+     echo "Using best checkpoint ==>  $CHECKPOINT  <=="
+  else
+     CHECKPOINT=$LAST_CHECKPOINT
+     echo "Using latest checkpoint ==>  $CHECKPOINT  <=="
+  fi
 else
   echo "Checkpoint supplied: ["${CHECKPOINT}"]"
 fi
 
 MODEL=`echo $CHECKPOINT |sed "s@^[^0-9]*\([0-9]\+\).*@\1@"`
-mkdir -p checkpoint
+rm -rf checkpoint
+cp -a upload-template checkpoint
+mkdir -p checkpoint/model
 MODEL_FILE=$MODEL_DIR"model_"$MODEL".pb"
 METADATA_FILE=$MODEL_DIR"model_metadata.json"
 
-
 if test ! -f "$MODEL_FILE"; then
     echo "$MODEL_FILE doesn't exist"
     exit 1
-else
-  cp $MODEL_FILE checkpoint/  
 fi
 
 if test ! -f "$METADATA_FILE"; then
     echo "$METADATA_FILE doesn't exist"
     exit 1
-else
-  cp $METADATA_FILE checkpoint/  
 fi
 
-CHECKPOINT_FILES=`echo $CHECKPOINT* |sed "s/\"//g"`
-for i in $( find $MODEL_DIR -type f -name $CHECKPOINT_FILES ); do
-  cp $i checkpoint/  
+cp -v $MODEL_FILE checkpoint/model/
+cp -v $METADATA_FILE checkpoint/model/
+
+CHECKPOINT_FILES=$MODEL_DIR/${CHECKPOINT}*
+#for i in $( find $MODEL_DIR -type f -name ${CHECKPOINT}\* ); do
+for i in $CHECKPOINT_FILES
+do
+  cp -v $i checkpoint/model/
 done
 
-VAR1=`cat $CHECKPOINT_FILE |jq ".last_checkpoint = .best_checkpoint"`
-VAR2=`echo $VAR1 |jq ".last_checkpoint.name = $CHECKPOINT"`
-VAR3=`echo $VAR2 |jq ".best_checkpoint.name = $CHECKPOINT"`
-echo $VAR3 >checkpoint/deepracer_checkpoints.json
+echo $CHECKPOINT > checkpoint/model/.coach_checkpoint
+# File deepracer_checkpoints.json is optional.
 
-# upload files to s3
-for filename in checkpoint/*; do
-  aws s3 cp $filename s3://$S3_BUCKET/$S3_PREFIX/model/
-done
+# Cleanup upload destination
+aws s3 rm --recursive s3://$S3_BUCKET/$S3_PREFIX/
```

Comment:
this one is removing all other folders which makes AWS unhappy, change it to: `aws s3 rm s3://$S3_BUCKET/$S3_PREFIX/model --recursive`

Reply:
I forgot to include the directory upload-template. So some files were missing and that was causing issues.

I would always clean up the whole prefix to ensure that only newly uploaded files are present. Otherwise, unknown old files could be affecting the import and your model.